tests/integration/anoncreds/bbs-plus-presentation.test.js

```diff
@@ -0,0 +1,183 @@
+    familyName: 'SMITH',
+    lprNumber: 1234,
+  },
+};
+
+describe('BBS+ Presentation', () => {
+  const dock = new DockAPI();
+  let bbsPlusPresentation;
+  let account;
+  let did1;
+  let pair1;
+  let chainModule;
+  let keypair;
+
+  beforeAll(async () => {
+    await initializeWasm();
+    await dock.init({
+      keyring: TestKeyringOpts,
+      address: FullNodeEndpoint,
+    });
+    chainModule = dock.bbsPlusModule;
+    account = dock.keyring.addFromUri(TestAccountURI);
+    dock.setAccount(account);
+    pair1 = dock.keyring.addFromUri(randomAsHex(32));
+    did1 = createNewDockDID();
+    await registerNewDIDUsingPair(dock, did1, pair1);
+  }, 20000);
+  beforeEach(() => {
+    bbsPlusPresentation = new BbsPlusPresentation(dock);
+  }, 20000);
+
+  test('Can create BBS+ public key for the DID', async () => {
+    keypair = Bls12381G2KeyPairDock2022.generate({
+      controller: did1,
+    });
+
+    const pk1 = BBSPlusModule.prepareAddPublicKey(u8aToHex(keypair.publicKeyBuffer));
+    await chainModule.addPublicKey(pk1, did1, did1, pair1, 1, { didModule: dock.did }, false);
+
+    const didDocument = await dock.did.getDocument(did1);
+    const { publicKey } = didDocument;
+
+    expect(publicKey.length).toEqual(2);
+    expect(publicKey[1].type).toEqual('Bls12381G2VerificationKeyDock2022');
+
+    keypair.id = publicKey[1].id;
+  }, 30000);
+
+  test('Can in add credentials to presentation builder', async () => {
+
+    const idx = await bbsPlusPresentation.addCredentialsToPresent(await getBBSCredential(did1, keypair));
+    expect(idx).toBe(0);
+  }, 30000);
+  test('expect to reveal specified attributes', async () => {
+
+    const idx = await bbsPlusPresentation.addCredentialsToPresent(await getBBSCredential(did1, keypair));
+    await bbsPlusPresentation.addAttributeToReveal(idx, ['credentialSubject.lprNumber']);
+    const presentation = await bbsPlusPresentation.createPresentation();
+    expect(presentation.spec.credentials[0].revealedAttributes).toHaveProperty('credentialSubject');
+    expect(presentation.spec.credentials[0].revealedAttributes.credentialSubject).toHaveProperty('lprNumber', 1234);
+  });
+  test('expect not to reveal any attributes', async () => {
+    await bbsPlusPresentation.addCredentialsToPresent(await getBBSCredential(did1, keypair));
+    const presentation = await bbsPlusPresentation.createPresentation();
+    expect(presentation.spec.credentials[0].revealedAttributes).toMatchObject({});
+  });
+  test('expect to throw exception when attributes provided is not an array', async () => {
+    const idx = await bbsPlusPresentation.addCredentialsToPresent(await getBBSCredential(did1, keypair));
+    expect(() => {
+      bbsPlusPresentation.addAttributeToReveal(idx, {});
+    }).toThrow();
+  });
+  afterAll(async () => {
+    await dock.disconnect();
+  }, 10000);
+});
+
+function getBBSCredential(did1, keypair) {
+  const credSchema = new CredentialSchema(residentCardSchema1, { useDefaults: true });
```

Comment:
updated in this branch, you should issue credentials similar to the issuing test https://github.com/docknetwork/sdk/blob/master/tests/integration/anoncreds/issuing.test.js#L118 

otherwise presentation being valid here may be a false positive